src/mapping.js

```diff
@@ -0,0 +1,212 @@
+import { decode } from 'sourcemap-codec';
+
+class GeneratedFragmentMapper {
+	constructor(
+		generated_code,
+		tag_info,
+	) {
+		this.generated_code = generated_code;
+		this.tag_info = tag_info;
+	}
+
+	get_position_relative_to_fragment(positionRelativeToFile) {
+        const fragment_offset = this.offset_in_fragment(offset_at(positionRelativeToFile, this.generated_code));
+        return position_at(fragment_offset, this.tag_info.generated_content);
+	}
+
+	offset_in_fragment(offset) {
+		return offset - this.tag_info.generated_start
+	}
+}
+
+class OriginalFragmentMapper {
+	constructor(
+		original_code,
+		tag_info,
+	) {
+		this.original_code = original_code;
+		this.tag_info = tag_info;
+	}
+
+	get_position_relative_to_file(positionRelativeToFragment) {
+		const parent_offset = this.offset_in_parent(offset_at(positionRelativeToFragment, this.tag_info.original_content));
+		return position_at(parent_offset, this.original_code);
+	}
+
+	offset_in_parent(offset) {
+		return this.tag_info.original_start + offset;
+	}
+}
+
+class SourceMapper {
+	constructor(raw_source_map) {
+		this.raw_source_map = raw_source_map;
+	}
+
+	getOriginalPosition(generated_position) {
```

Comment:
It's making me uneasy that the tests still passed even though the code in `preprocess.ts` has been changed to use `get_original_position` while this is still `getOriginalPosition`.

Reply:
This is a method called in line 126, so it's a different method (which I forgot to convert to _-style)

Reply:
Ah, okay, thanks.